ci/test-clang-docker.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ ! -d /repo ]]; then
+    echo "[-] This script needs to be run within Docker! Use test-clang.sh instead."
+    exit 1
+fi
+
+clang=/repo/toolchain/bin/clang
+hello_c=/repo/llvm-project/hello.c
+hello_cpp=/repo/llvm-project/hello.cpp
+hello_exe=/tmp/hello
+host_arch=$(uname -m)
+
+# Print clang information, which makes sure it can run
+echo "[+] Testing 'clang --version'"
+"$clang" --version
+
+case "$(source /usr/lib/os-release; echo "$ID")" in
+    fedora)
+        musl_cc_flags=(
+            -B /usr/"$host_arch"-linux-musl/lib64   # Scrt1.o, crti.o, crtn.o
+            -I /usr/"$host_arch"-linux-musl/include # stdio.h
+            -L /usr/"$host_arch"-linux-musl/lib64   # -lc
+        )
+
+        echo "[+] Updating OS"
+        dnf update -qy
+
+        echo "[+] Installing musl"
+        dnf install -qy \
+            musl-devel \
+            musl-libc \
+            musl-libc-static
+        ;;
+esac
+
+########
+# MUSL #
+########
+
+echo "[+] Testing dynamically linking hello.c against musl"
+"$clang" "${musl_cc_flags[@]}" -o "$hello_exe" "$hello_c"
+"$hello_exe"
+
+echo "[+] Testing statically linking hello.c against musl"
+"$clang" "${musl_cc_flags[@]}" -static -o "$hello_exe" "$hello_c"
+"$hello_exe"
+
+echo "[+] Testing dynamically linking hello.cpp against musl"
+# --rpath to avoid having to pull in libc++ and libunwind from distribution
+"$clang"++ "${musl_cc_flags[@]}" -Wl,--rpath=/repo/toolchain/lib/"$host_arch"-alpine-linux-musl -o "$hello_exe" "$hello_cpp"
```

Comment:
In addition to rpath, can you add another compilation that doesn't use rpath, but does use `LD_LIBRARY_FLAGS` when running the binary?

cc @compnerd @MaskRay

Reply:
I assume you mean `LD_LIBRARY_PATH`? I did try that but I couldn't really get it to work. I'll have another go at it tomorrow, as I was kind of racing to get something up for review today.

Reply:
Yes (sorry).  If it still doesn't work, I'd be curious the error.

Reply:
Done: 179aaa1399e18c519f66fb0eca20ee849a3e97b0